### PR TITLE
Update step `I pick a random machineset to scale`

### PR DESCRIPTION
**Pull request by sunzhaohua2**
Update step `I pick a random machineset to scale` in case windows machineset is selected in a cluster with windows nodes. @jhou1 @miyadav please help to take a look.

### CODE REVIEW

**Comment by miyadav** [comment]
LGTM , shall we run it once and confirm the results ?

**Comment by sunzhaohua2** [comment]
thanks for reminder @miyadav I tested clone a linux machineset with case OCP-28108: 
```
      [05:50:41] INFO> === End After Scenario: Cluster should automatically scale up and scale down with clusterautoscaler deployed ===

1 scenario (1 passed)
28 steps (28 passed)
12m7.297s
```
 I tested clone a windows machineset with steps:
```
  Scenario: Test create a windows machineset
    Given I have an IPI deployment
    And I switch to cluster admin pseudo user
    And I use the "openshift-machine-api" project
    And admin ensures machine number is restored after scenario

    Given I clone a windows machineset and name it "machineset-windows"
```
clone a windows machineset result:
```
      [07:25:23] INFO> === End After Scenario: Test create a windows machineset ===

1 scenario (1 passed)
5 steps (5 passed)
9m34.887s
```

File: features/step_definitions/machine_set.rb [hunk 1]
```diff
@@ -1,7 +1,14 @@
 Given(/^I pick a random machineset to scale$/) do
   ensure_admin_tagged
   machine_sets = BushSlicer::MachineSet.list(user: admin, project: project("openshift-machine-api")).
-    select { |ms| ms.available_replicas >= 1 }
+    select { |ms| ms.available_replicas >= 1 && ms.windows_machinesets != "Windows"}
+  cache_resources *machine_sets.shuffle
+end
+
+Given(/^I pick a random windows machineset to scale$/) do
```
Comment (liangxia) [inline features/step_definitions/machine_set.rb:8]: `Given(/^I pick a random (windows|linux) machineset to scale$/)`

File: lib/openshift/machine_set.rb [hunk 2]
```diff
@@ -40,5 +40,10 @@ def taints(user: nil, cached: true, quiet: false)
       raw_resource(user: user, cached: cached, quiet: quiet).
         dig('spec', 'taints')
     end
+
+    def windows_machinesets(user: nil, cached: true, quiet: false)
```
Comment (liangxia) [inline lib/openshift/machine_set.rb:44]: Anything indicate it's a windows machineset ?

Comment (jhou1) [reply lib/openshift/machine_set.rb:44]: The naming and definition of the method can be more straightforward, I suggest defining a method like `is_windows_machineset?` which returns true or false.

```
def is_windows_machineset?(user: nil, cached: true, quiet: false)
  rr = raw_resource(user: user, cached: cached, quiet: quiet)
  rr.dig('spec', 'template', 'metadata', 'labels','machine.openshift.io/os-id') == "Windows"
end
```

File: features/step_definitions/machine_set.rb (hunk 1 above)
Comment (jhou1) [reply features/step_definitions/machine_set.rb:8]: The original step can be updated slimilar to `I pick a random( windows)? machineset to scale` with condition to deal with windows machine scenario.

File: lib/openshift/machine_set.rb (hunk 2 above)
Comment (akostadinov) [reply lib/openshift/machine_set.rb:44]: @jhou1 , I fully agree with you. Also related to what we spoke about earlier.

**Comment by sunzhaohua2** [comment]
@miyadav @jhou1 @liangxia @akostadinov have updated and tested locally, it passed, please help to review again.

File: features/step_definitions/machine_set.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
-Given(/^I pick a random machineset to scale$/) do
+Given(/^I pick a random( windows)? machineset to scale$/) do | windows |
   ensure_admin_tagged
   machine_sets = BushSlicer::MachineSet.list(user: admin, project: project("openshift-machine-api")).
-    select { |ms| ms.available_replicas >= 1 }
+    select { |ms| ms.available_replicas >= 1 && (!windows || ms.is_windows_machinesets?) }
```
Comment (jhou1) [inline features/step_definitions/machine_set.rb:4]: What about `(windows ? ms.is_windows_machinesets? : !ms.is_windows_machinesets?)`

**Pull request merged**